src/ApiException.php

```diff
@@ -151,13 +247,21 @@ public static function createFromRestApiResponse(
      */
     private static function create($basicMessage, $rpcCode, $metadata, array $decodedMetadata, $previous = null)
     {
+        $containsErrorInfo = self::containsErrorInfo($decodedMetadata);
         $rpcStatus = ApiStatus::statusFromRpcCode($rpcCode);
         $messageData = [
             'message' => $basicMessage,
             'code' => $rpcCode,
             'status' => $rpcStatus,
             'details' => $decodedMetadata
         ];
+        if ($containsErrorInfo['containsErrorInfo']) {
+            $messageData = array_merge([
+                'domain' => $containsErrorInfo['domain'],
+                'reason' => $containsErrorInfo['reason'],
+                'errorInfoMetadata' => $containsErrorInfo['errorInfoMetadata'],
+            ], $messageData);
```

Comment:
Nit: Alternatively, you could `unset($containsErrorInfo['containsErrorInfo'])` and then `array_merge($containsErrorInfo, $messageData)`

Reply:
that's neat - done!